### PR TITLE
 ⭐️ add aws.elasticache.cacheClusters resource, deprecate elasticache.clusters

### PR DESCRIPTION
**Pull request by vjeffrey**
```
cnquery> aws.elasticache.cacheClusters { * }
aws.elasticache.cacheClusters: [
  0: {
    arn: "arn:aws:elasticache:us-east-1:ID:cluster:vjtest-001"
    notificationConfiguration: ""
    cacheSubnetGroupName: "vjtest"
    atRestEncryptionEnabled: false
    logDeliveryConfigurations: []
    ipDiscovery: "ipv4"
    authTokenLastModifiedDate: null
    numCacheNodes: 1
    engine: "redis"
    networkType: "ipv4"
    region: "us-east-1"
    securityGroups: []
    transitEncryptionMode: ""
    cacheNodeType: "cache.t3.medium"
    engineVersion: "7.0.7"
    preferredAvailabilityZone: "us-east-1c"
    transitEncryptionEnabled: false
    cacheClusterId: "vjtest-001"
    authTokenEnabled: false
    snapshotRetentionLimit: 0
```

### CODE REVIEW

File: resources/packs/aws/aws.lr [hunk 1]
```diff
@@ -1186,7 +1186,67 @@ private aws.rds.dbinstance @defaults("arn") {
 // Amazon ElastiCache
 aws.elasticache {
   // List of clusters
-  clusters() []dict
+  // clusters() []dict // deprecated
+  // List of clusters
+  cacheClusters() []aws.elasticache.cluster
+}
+
+// Amazon Elasticache Cluster
```
Comment (tas50) [inline resources/packs/aws/aws.lr]:
```suggestion
// Amazon ElastiCache Cluster
```

File: resources/packs/aws/aws.lr [hunk 2]
```diff
@@ -1186,7 +1186,67 @@ private aws.rds.dbinstance @defaults("arn") {
 // Amazon ElastiCache
 aws.elasticache {
   // List of clusters
-  clusters() []dict
+  // clusters() []dict // deprecated
```
Comment (vjeffrey) [inline resources/packs/aws/aws.lr]: need to bring this back in, will push commit shortly

**Comment by scottford-io** [comment]
@vjeffrey Thank you so much for getting to this. I few items of note ...

I noticed that `help aws.elasticache.cacheClusters` does not work in the shell:

```
cnquery> help aws.elasticache
aws.elasticache:                           Amazon ElastiCache
  cacheClusters []aws.elasticache.cluster: List of clusters
  clusters []dict:                         Deprecated. Use `cacheClusters` instead.
cnquery> help aws.elasticache.cacheClusters
cnquery>
```

Second, I also noticed that querying `aws.elasticache {*}` in the shell still returns that full data object. I am not sure if this is by design or not. 🤔

**Review by scottford-io — APPROVED** [review]
@vjeffrey I am going to approve, but left a couple of comments regarding usage in the shell.

**Comment by vjeffrey** [comment]
> help aws.elasticache.cacheClusters

it looks like help only works on the top level resource:
```
cnquery> help aws.s3
aws.s3:                    Amazon S3 Cloud Object Storage
  buckets []aws.s3.bucket: List of S3 buckets across the account
cnquery> help aws.elasticache
aws.elasticache:                           Amazon ElastiCache
  cacheClusters []aws.elasticache.cluster: List of clusters
  clusters []dict:                         Deprecated. Use `cacheClusters` instead.
cnquery> help aws.s3.buckets
cnquery>
```

**Comment by vjeffrey** [comment]
> aws.elasticache {*} in the shell still returns that full data object.

ya, i've marked clusters() as deprecated, but it still exists/fetches information. we'll get that removed in the future once the deprecation has been out there for a bit

**Pull request merged**